src/uct/sm/mm/mm_iface.h

```diff
@@ -9,20 +9,93 @@
 #define UCT_MM_IFACE_H
 
 #include <uct/tl/tl_base.h>
+#include <ucs/debug/memtrack.h>
+#include <sys/shm.h>
+#include "mm_def.h"
+#include "mm_pd.h"
 
 #define UCT_MM_TL_NAME "mm"
 
+#define UCT_MM_FIFO_CTL_SIZE_ALIGNED    ucs_align_up(sizeof(uct_mm_fifo_ctl_t),UCS_SYS_CACHE_LINE_SIZE)
 
 typedef struct uct_mm_iface_config {
-    uct_iface_config_t      super;
+    uct_iface_config_t       super;
+    unsigned                 fifo_size;            /* Size of the receive FIFO */
+    double                   release_fifo_factor;
+    ucs_ternary_value_t      hugetlb_mode;         /* Enable using huge pages for */
+                                                   /* shared memory buffers */
+    uct_iface_mpool_config_t mp;
 } uct_mm_iface_config_t;
 
 
-typedef struct uct_mm_iface {
+struct uct_mm_iface {
     uct_base_iface_t        super;
-    /* TODO shared FIFO id and address */
-} uct_mm_iface_t;
 
+    /* Receive FIFO */
+    uct_mm_id_t             fifo_mm_id;       /* memory id which will be received */
+                                              /* after allocating the fifo */
+    void                    *recv_fifo;       /* the beginning of the receive fifo */
+
+    uct_mm_fifo_ctl_t       *recv_fifo_ctl;   /* pointer to the struct at the */
+                                              /* beginning of the receive fifo */
+                                              /* which holds the head and the tail */
+    uct_mm_fifo_element_t   *recv_fifo_elements; /* pointer to the first fifo element */
```

Comment:
it should be `void *` (same as on the sender side - mm_ep)  - because it's not an array, elements also have data, and besides, you always cast it to 'char*' anyway